src/pages/search.vue

```diff
@@ -26,34 +27,50 @@ import {
   SET_SEARCH_STATE_FROM_URL,
   UPDATE_SEARCH_TYPE,
 } from '~/constants/action-types'
-import { SET_FILTER_IS_VISIBLE } from '~/constants/mutation-types'
 import { queryStringToSearchType } from '~/utils/search-query-transform'
 import local from '~/utils/local'
 import { ALL_MEDIA, IMAGE, VIDEO } from '~/constants/media'
-import { mapActions, mapGetters, mapMutations, mapState } from 'vuex'
+import { mapActions, mapGetters, mapState } from 'vuex'
 import { MEDIA, SEARCH } from '~/constants/store-modules'
 import debounce from 'lodash.debounce'
-import SearchGridFilter from '~/components/Filters/SearchGridFilter.vue'
+
+import { onMounted } from '@nuxtjs/composition-api'
 import { useMediaQuery } from '~/composables/use-media-query.js'
+import { useUiState } from '~/composables/use-ui-state'
+
 import AppModal from '~/components/AppModal'
+import VSearchGridFilter from '~/components/VFilters/VSearchGridFilter.vue'
+import VFilterDisplay from '~/components/VFilters/VFilterDisplay.vue'
 
 const MIN_SCREEN_WIDTH_FILTER_VISIBLE_BY_DEFAULT = 768
 
 const BrowsePage = {
   name: 'browse-page',
   layout: 'default',
   components: {
-    SearchGridFilter,
+    VFilterDisplay,
+    VSearchGridFilter,
   },
   setup() {
     const defaultWindow = typeof window !== 'undefined' ? window : undefined
     const isLargeScreen = useMediaQuery(
       `(min-width: ${MIN_SCREEN_WIDTH_FILTER_VISIBLE_BY_DEFAULT}px)`,
       defaultWindow
     )
+    const { isFilterSidebarVisible, setFilterSidebarVisibility } = useUiState()
 
+    onMounted(() => {
+      const localFilterState = () =>
+        local.get(process.env.filterStorageKey)
+          ? local.get(process.env.filterStorageKey) === 'true'
+          : true
+      setFilterSidebarVisibility(isLargeScreen && localFilterState())
+    })
```

Comment:
I wonder if this could all be built into the `useUiState` composable? In fact, to avoid `useUiState` from growing to encompass the entire UI state, could we rename it to `useFilterSidebarVisibility` or something like that? And then all the logic for determining whether it is open or closed etc can go in there.

Reply:
I had an idea to use `UI state` for state that is mainly about the UI (something shown or not), but you are right, it can grow uncontrollably, so I renamed it.